src/sqlfluff/core/linter/linting_result.py

```diff
@@ -114,7 +114,7 @@ def get_violations(self, **kwargs):
     def violation_dict(self, **kwargs):
         """Return a dict of paths and violations."""
         return self.combine_dicts(
-            path.violation_dict(**kwargs) for path in self.paths
+            *(path.violation_dict(**kwargs) for path in self.paths)
         )  # pragma: no cover TODO?
```

Comment:
I don't think we need to add the `*()` it just adds complexity. I also reckon we can add the `no cover` tag too now you've added the additional test below. I'll make that tweek and then get this merged 👍